var/spack/repos/builtin/packages/rayleigh/package.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack.package import *
+
+
+class Rayleigh(MakefilePackage):
+    __doc__ = """Rayleigh is a 3-D convection code designed for the study of
+    dynamo behavior in spherical geometry."""
+    homepage = "https://github.com/geodynamics/Rayleigh"
+    url = "https://github.com/geodynamics/Rayleigh/archive/refs/tags/v1.0.1.tar.gz"
+    git = "https://github.com/geodynamics/Rayleigh.git"
+    maintainers = ["tukss"]
+    version("main", branch="main")
+    version("1.1.0", sha256="93fbbdbde6088807638e4dcbd4d622203fd4753c1831bab2cb8eaeca5cba45c3")
+    version("1.0.1", sha256="9c9e3b0b180f32a889f158e2ea2967f4ac2bb2124f5d264f230efb8c8f19ea36")
+    version("1.0.0", sha256="4f2e8249256adff8c4b0bc377ceacf8a6441dcee54b7d36c784f05a454dc6dcf")
+    version("0.9.1", sha256="ab96445fc61822fe2d2cba8729a85b36de6b541febf5759de6d614847844573f")
+    version("0.9.0", sha256="63a80d1619cb639f3cb01ab82a441b77d736eee94469c47c50ab740fa81c08f4")
+    depends_on("mpi")
+    depends_on("fftw-api@3")
+    depends_on("lapack")
+
+    def edit(self, spec, prefix):
```

Comment:
I think the `edit` function should be two functions in this case:
- `setup_build_environment` for setting the `FC` environment variable,
- `configure_args` for the args to configure.

There should be no reason to call `configure` explicitly in the package; `configure_args` should define the arguments (minus prefix).

Reply:
I've split it into three functions now. It looks like I still need to invoke `configure` manually, otherwise it is not called at all. Note that this is a simple `MakefilePackage` and not an `AutotoolsPackage`. It might look like it because it has a `configure` script but this is actually a hand-crafted script that has nothing to do with autotools and friends.

Reply:
I see. If this is just easier with `configure`, then I am also happy to approve it that way. Maybe add some comments so the next reviewer (on a new version PR) doesn't suggest the same thing.